consensus/core/src/sign.rs

```diff
@@ -22,3 +54,157 @@ pub fn sign(mut signable_tx: SignableTransaction, schnorr_key: secp256k1::KeyPai
     }
     signable_tx
 }
+
+/// Sign a transaction using schnorr
+pub fn sign_with_multiple(mut mutable_tx: SignableTransaction, privkeys: Vec<[u8; 32]>) -> SignableTransaction {
+    let mut map = BTreeMap::new();
+    for privkey in privkeys {
+        let schnorr_key = secp256k1::KeyPair::from_seckey_slice(secp256k1::SECP256K1, &privkey).unwrap();
+        map.insert(schnorr_key.public_key().serialize(), schnorr_key);
+    }
+    for i in 0..mutable_tx.tx.inputs.len() {
+        mutable_tx.tx.inputs[i].sig_op_count = 1;
+    }
+
+    let mut reused_values = SigHashReusedValues::new();
+    for i in 0..mutable_tx.tx.inputs.len() {
+        let script = mutable_tx.entries[i].as_ref().unwrap().script_public_key.script();
+        if let Some(schnorr_key) = map.get(script) {
+            let sig_hash = calc_schnorr_signature_hash(&mutable_tx.as_verifiable(), i, SIG_HASH_ALL, &mut reused_values);
+            let msg = secp256k1::Message::from_slice(sig_hash.as_bytes().as_slice()).unwrap();
+            let sig: [u8; 64] = *schnorr_key.sign_schnorr(msg).as_ref();
+            // This represents OP_DATA_65 <SIGNATURE+SIGHASH_TYPE> (since signature length is 64 bytes and SIGHASH_TYPE is one byte)
+            mutable_tx.tx.inputs[i].signature_script = std::iter::once(65u8).chain(sig).chain([SIG_HASH_ALL.to_u8()]).collect();
+        }
+    }
+    mutable_tx
+}
+
+/// Sign a transaction using schnorr
+pub fn sign_with_multiple_v2(mut mutable_tx: SignableTransaction, privkeys: Vec<[u8; 32]>) -> SignableTransaction {
```

Comment:
what does `v2` stand for?
can we remove the commented log lines all over this module?

Reply:
In a few places, I did not want us to "blanket update" consensus functions and defer merging/fixing them in a separate PR.  The original function has a problem (have to look at what it is), so I want to rectify it separately (as this PR as it is, is very large).  This has been the general policy for a few elements (mass calc, network id, sign_with_multiple, rpc hex utils being moved to kaspa-utils, and probably a few others).

Reply:
so, current `sign_with_multiple` forces all sigop counts to `1`.  I suggest the merge of the original and v2 is done in a separate PR as I don't know why something like this was done in the signing fn...